### ISSUE TITLE
Confusion about the results of `boot_genes` in PLS analysis

### ISSUE THREAD

**Issue by LuuuXG** [alegiac95/Imaging-transcriptomics#29]
In `pls.py`, lines 148-151:

![image](https://github.com/user-attachments/assets/99f304b4-e57e-4d4d-b488-362cb284670a)

I am confused about why it is `orig.` for genes and z-score, but `boot.` for the p-value and adjusted p-value. As in `genes.py`, within the `compute()` function, `_p_val` and `_p_corr` seem to be ordered according to `_index`, which is the order of `_z`, rather than the original order. Although the two orders are similar, that is how the bootstrap process works.

In other words, in the output file `pls_component_*.tsv`, the order of the genes is the SAME as their order from the initial PLS regression result (in `genes.py`, line 120). This is really confusing.

Additionally, in `genes.py`, within the `gsea` function, the `orig.genes` is again used. If this is the case, why do we need to conduct the bootstrap of genes and get `boot.genes`?

I would appreciate any clarification on this matter.
